### ISSUE TITLE
Lights locking up or not working as expected.

### ISSUE THREAD

**Issue by J1pp** [pimoroni/keybow-python#2]
See issue with lights locking up or not always responding as expected to show function.

Issue found to be with show function in __init__.py

line 
`_buf += [0b00000000 for _ in range(1)]`
should be 
`_buf += [0b11111111 for _ in range(1)]`

**Comment by Gadgetoid** [comment]
Might have to look into this- these APA102 parts seem to have variants that work in unpredictable and incompatible ways. Need to double check this has always been broken, versus suddenly happening with a newer Keybow unit. Do you have any test code?

**Comment by J1pp** [comment]
Below is some test code that shows the problem as coded. Sorry this is not formatting well'
```python
import spidev
import time
import random

spi = spidev.SpiDev() # create spi object
spi.open(0, 0) # open spi port 0, device (CS) 1
spi.mode = 0b00
spi.max_speed_hz = 4000000
spi.open(0, 0) # open spi port 0, device (CS) 0

try:
    while True:
        for _ in range(10):
            spi.writebytes([0, 0, 0, 0, 0b11100011, 255, 0, 0, 0b11100011, 0, 255, 0, 0b11100011, 0, 0, 255])
            time.sleep(0.5) # sleep for 0.1 seconds
            spi.writebytes([0, 0, 0, 0, 0b11100011, 0, 0, 0, 0b11100011, 0, 0, 0, 0b11100011, 0, 0, 0, 255])
            time.sleep(0.5)
        for _ in range(10):
            spi.writebytes([0, 0, 0, 0, 0b11100011, 255, 0, 0, 0b11100011, 0, 255, 0, 0b11100011, 0, 0, 0])
            time.sleep(0.5) # sleep for 0.1 seconds
            spi.writebytes([0, 0, 0, 0, 0b11100011, 0, 0, 0, 0b11100011, 0, 0, 0, 0b11100011, 0, 0, 0, 0])
            time.sleep(0.5)

except KeyboardInterrupt: # Ctrl+C pressed, so ...
    spi.writebytes([0, 0, 0, 0, 0b11100011, 0, 0, 0, 0b11100011, 0, 0, 0, 0b11100011, 0, 0, 0, 255, 255, 255, 255]) # b, g, r
    spi.close() # ... close the port before exit
```

This seems to relate to the issue raised in the firware section 
[https://github.com/pimoroni/keybow-firmware/issues/23](url)
Lack of documentation: Why 0b11100011? #23
And the protocol posted by yourself
which details the end frame needing to be all 1's
however in init.py the end frame is set to all 0's for the show() function. Again not formatting well sorry.

`def show():
    """Update LEDs on Keybow."""
    setup()
    # Start of frame, 4 empty bytes
    _buf = [0b00000000 for _ in range(8)]
    for rgbbr in buf:
        r, g, b, br = rgbbr
        br = int(br * 31)
        _buf.append(0b11100000 | br)   # Start of LED frame, 0b11100000 + brightness
        _buf.append(b)
        _buf.append(g)
        _buf.append(r)
    # End of frame, 4 empty bytes
    _buf += [0b00000000 for _ in range(1)]
    spi.xfer2(_buf)`

**Comment by Gadgetoid** [comment]
If you use three backticks you should be able to apply multiline formatting. Have updated your comment to format the first code snippet by way of example.

**Comment by Gadgetoid** [comment]
Tested and confirmed on a 12-key keybow using:

```python
import spidev
import time

spi = spidev.SpiDev()
spi.open(0, 0)
spi.mode = 0b00
spi.max_speed_hz = 4000000
spi.open(0, 0)

sof = [0, 0, 0, 0]
r = [0b11100011, 255, 0, 0]
g = [0b11100011, 0, 255, 0]
b = [0b11100011, 0, 0, 255]
off = [0b11100011, 0, 0, 0]
eof = [255, 255, 255, 255]

try:
    while True:
        for eof in ([255, 255, 255, 255], [0, 0, 0, 0]):
            print(eof)
            for _ in range(10):
                output = sof + ((r + g + b) * 4) + eof
                spi.writebytes(output)
                time.sleep(0.1)
                spi.writebytes(sof + (off * 12) + eof)
                time.sleep(0.1)

except KeyboardInterrupt:
    spi.writebytes(sof + (off * 12) + eof)
    spi.close()
```

with an EOF of `0, 0, 0, 0` it's completely bonkers, but with `255, 255, 255, 255` it appears fine.

**Comment by Gadgetoid** [comment]
Should hopefully be fixed in release `0.0.3`
